### PR TITLE
TFBart lables consider both pad token and -100

### PR DESCRIPTION
**Pull request by kiyoungkim1**
For #9770,

1. ```TFBartModels``` use -100 as a masking token for ```decoder_input_ids``` and ```compute_loss``` like other models(```T5```).
2. For legacy, all the  ```padding token``` in ```labels``` are replace by ```-100`` token.

Below examples show the same result for ```labels``` with ```-100 token``` or ```padding token```, where previously ```Nan``` are shown in the latter case(#9770). 

```
import tensorflow as tf
from transformers import BartTokenizer, TFBartForConditionalGeneration

tokenizer = BartTokenizer.from_pretrained("facebook/bart-base")
model = TFBartForConditionalGeneration.from_pretrained("facebook/bart-base")

inputs = tokenizer("My dog is <mask>", return_tensors='tf', truncation=True, max_length=16, padding="max_length")
labels_ids = tokenizer("My dog is cute", return_tensors='tf', truncation=True, max_length=16, padding="max_length").input_ids

## labels padding_token = 1
loss = model(inputs, labels=labels_ids)[0]
print(labels_ids)
print(loss)

## labels padding_token = -100
labels_ids = tf.where(
    labels_ids == 1, tf.fill(tf.shape(labels_ids), tf.constant(-100, dtype='int32')), labels_ids
)

loss = model(inputs, labels=labels_ids)[0]
print(labels_ids)
print(loss)
```

```
tf.Tensor(
[[    0  2387  2335    16 11962     2     1     1     1     1     1     1
      1     1     1     1]], shape=(1, 16), dtype=int32)
tf.Tensor(
[2.2291888e-05 4.8874615e-05 3.7192607e-05 7.9230859e-04 6.1941862e+00
 1.1058818e+00], shape=(6,), dtype=float32)
tf.Tensor(
[[    0  2387  2335    16 11962     2  -100  -100  -100  -100  -100  -100
   -100  -100  -100  -100]], shape=(1, 16), dtype=int32)
tf.Tensor(
[2.2291888e-05 4.8874615e-05 3.7192607e-05 7.9230859e-04 6.1941862e+00
 1.1058818e+00], shape=(6,), dtype=float32)
```

TFBart gives the same result with both -100 and padding token.
However, ```Bart(pytorch) with -100 token in labels```, ```Bart with padding token in labels``` and ```TFBart(tensorflow with -100 or padding token)``` gives three different results. This is noticed but not treated in this PR.

@patrickvonplaten
@jplu 
@patil-suraj

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
I like this very much! The code looks fully backward compatible to me since in labels `self.config.pad_token_id` is replaced by -100! Very nice clean up @kiyoungkim1 :-)

Could you quickly rebase this PR to master (or merge master into the PR) so that the circle ci is not red anymore? 

Great work!

**Review by jplu — APPROVED** [review]
LGTM! I'm fine with this fix, thanks a lot for taking care of it ^^

**Review by sgugger — APPROVED** [review]
Thanks for your PR!

**Comment by kiyoungkim1** [comment]
@patrickvonplaten 
I merged upstream to the branch!

**Review by LysandreJik — APPROVED** [review]
LGTM

**Comment by LysandreJik** [comment]
You have an error in code quality, could you run `make style` and `make quality` to check it out? Thanks.
